core/src/main/java/org/elasticsearch/action/DocWriteResponse.java

```diff
@@ -109,6 +110,22 @@ public RestStatus status() {
         return getShardInfo().status();
     }
 
+    /**
+     * Gets the location of the written document as a string suitable for a {@code Location} header.
+     * @param routing any routing used in the request. If null the location doesn't include routing information.
+     */
+    public String getLocation(@Nullable String routing) {
+        // Absolute path for the location of the document. This should be allowed as of HTTP/1.1:
```

Comment:
I'm not sure how to express this without writing a little paragraph comment.

In HTTP/1.0 it had to be an absolute URL. In HTTP/1.1 it is allowed to be a relative URL, resolved against the URL the request was received from. The spec specifically mentioned fun stuff like `Location: ../grumpy/cat/pictures`. My understanding is that absolute paths work as well. I expect even protocol relative paths would be acceptable.

So: upshot is that the *absolute* path is a *relative* URL and that is allowed in HTTP/1.1. I'm fairly sure.